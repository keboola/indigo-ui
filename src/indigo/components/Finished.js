import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';

class Finished extends React.Component {
  render() {
    return (
      <span
        title={
          this.props.endTime ? moment(this.props.endTime).format(this.props.dateTimeFormat) : ''
        }
      >
        {!!this.props.hasIcon && <i className="fa fa-calendar" />}
        {this.props.endTime ? moment(this.props.endTime).fromNow() : 'N/A'}
      </span>
    );
  }
}

Finished.propTypes = {
  endTime: PropTypes.string,
  dateTimeFormat: PropTypes.string,
  hasIcon: PropTypes.bool,
};

Finished.defaultProps = {
  dateTimeFormat: 'YYYY-MM-DD HH:mm:ss',
  hasIcon: false,
};

export default Finished;
