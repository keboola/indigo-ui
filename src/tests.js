import renderer from 'react-test-renderer';

export const snapshot = (component) => {
  const tree = renderer
    .create(component)
    .toJSON();
  expect(tree).toMatchSnapshot();
};
