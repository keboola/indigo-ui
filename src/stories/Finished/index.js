import React from 'react';
import {storiesOf} from '@storybook/react';
import {withInfo} from '@storybook/addon-info';

import Finished from '../../indigo/components/Finished';

const demoTimestamp = 1542632032608
storiesOf('Finished', module).add(
  'Basic Init',
  withInfo({
    text: `
        Shows how much time ago event occured. Shows exact time of event on mouse hover
        `,
    inline: true,
  })(() => {
    return <Finished endTime={demoTimestamp}/>;
  })
).add(
  'Finished with icon',
  withInfo({
    inline: true,
  })(() => {
    return <Finished hasIcon={true} endTime={demoTimestamp}/>;
  })
).add(
  'No datetime provided',
  withInfo({
    inline: true,
  })(() => {
    return <Finished/>;
  })
);
