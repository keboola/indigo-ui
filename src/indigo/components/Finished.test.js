import React from 'react';
import moment from 'moment';

import { snapshot } from '../../tests';
import Finished from "./Finished";

const DATE_TO_USE = new Date('2016');
const _Date = Date;
global.Date = jest.fn(() => DATE_TO_USE);
global.Date.UTC = _Date.UTC;
global.Date.parse = _Date.parse;
global.Date.now = _Date.n
const timeMock = moment(DATE_TO_USE).format('YYYY-MM-DD HH:mm:ss').toString()

describe('<Finished />', () => {
  it('Basic Init', () => {
    snapshot(<Finished endTime={timeMock}/>);
  });

  it('Finished with icon', () => {
    snapshot(<Finished showIcon endTime={timeMock}/>);
  });

  it('No datetime provided', () => {
    snapshot( <Finished/>);
  });
});
