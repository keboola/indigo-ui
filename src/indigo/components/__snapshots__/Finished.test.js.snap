// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Finished /> Basic Init 1`] = `
<span
  title="2016-01-01 01:00:00"
>
  a few seconds ago
</span>
`;

exports[`<Finished /> Finished with icon 1`] = `
<span
  title="2016-01-01 01:00:00"
>
  <i
    className="fa fa-calendar finished-icon"
  />
  a few seconds ago
</span>
`;

exports[`<Finished /> No datetime provided 1`] = `
<span
  title=""
>
  N/A
</span>
`;
